pint_pandas/pint_array.py

```diff
@@ -216,7 +241,8 @@ def __repr__(self):
     np.complex64: _NumpyEADtype("complex64"),
     # np.float16: pd.Float16Dtype(),
 }
-dtypeunmap = {v: k for k, v in dtypemap.items()}
+ddtypemap: dict[np.dtype, object] = {np.dtype(k): v for k, v in dtypemap.items()}
```

Comment:
what's this needed for?

Reply:
it is used line 300: `if dtype in ddtypemap:` where `dtype` is actually some `np.dtype("xxxx")` variable. 

Without this `ddtypemap` the test was failing and line 302 `pd.array` was fed with unrelevant dtype argument and was leading for all dtype to `NumpyExtensionArray`.
Now that `ddtypemap` is introduced `pd.array` retruns Int64Dtype, Float64Dtype, or any other ExtensionArray